docs/active_storage_migration.md

```diff
@@ -0,0 +1,387 @@
+
+```ruby
+rails> srs = Stats::StatsReport.all
+
+# Total number of stats reports
+rails> srs.count
+
+# Number of stats reports with nil checksum
+rails> srs.where(as_attachment_checksum: nil).count
+# Should be zero
+
+# Number of stats reports with a valid checksum
+# A valid checksum ends with '==', such as '4+g5Nmjax5aCg3HtRmCO5Q=='
+rails> srs.select { |sr| sr.as_attachment_checksum&.match(/==$/) }.count
+# Should be the same as srs.count, above
+```
+
+3) Migrate assets details from Paperclip to Active Storage:
+
+```bash
+$ bundle exec rails 'storage:migrate[stats_reports]'
+```
+
+4) Confirm that all reports have been migrated:
+
+```ruby
+# List of Paperclip filenames
+# Note that there may be duplicate Stats::StatsReport records for the same
+# file but the file will only be copied to Active Storage once
+rails> pc_filenames = Stats::StatsReport.pluck(:document_file_name).uniq.sort
+
+# List of Active Storage filenames
+rails> as_filenames = ActiveStorage::Attachment.where(record_type: 'Stats::StatsReport').map(&:filename).map(&:to_s).sort
+
+rails> pc_filenames == as_filenames
+# => true inducates that the migration is successful
+# => false indicates that the migraiton has failed
+```
+
+5) Test that a sample document can be downloaded via Active Storage (this will only work when S3 is used for storage):
+
+```ruby
+# Get a download url for a sample document
+rails> ActiveStorage::Attachment.where(record_type: 'Stats::StatsReport').sample.service_url
+# => "https://cloud-platform-..."
+# Paste this url into a browser and confirm that it can be downloaded
+# Note that this url will expire in 5 minutes
+```
+
+6) (Final step) Activate Active Storage for stats reports by deploying PR #3722 (CBO-1683)
+
+### Testing
+
+1) After the final step above has been completed the first test is to download
+   the reports at `case_workers/admin/management_information` without
+   regenerating.
+2) Next, regenerate the report and download again. Confirm that the new
+   document is different from the previous.
+
+### Rollback
+
+**Note 1:** This rollback should not be attempted after PR #3722 has been deployed
+(final step, above) as this will delete assets from Active Storage that do not
+exist in Paperclip.
+
+**Note 2:** Using `destroy` or `destroy_all` on `ActiveStorage::Attachment` may
+delete the actual files so `delete` is used instead, even though it is more
+involved.
+
+1) Get a list of stats reports records in Active Storage:
+
+```ruby
+rails> attachments = ActiveStorage::Attachment.where(record_type: 'Stats::StatsReport')
+```
+
+2) Record the list of blobs for these attachments:
+
+```ruby
+rails> blobs = ActiveStorage::Blob.where(id: attachments.pluck(:blob_id))
+```
+
+3) Remove attachment and blob records:
+
+```ruby
+rails> attachments.delete_all
+rails> blobs.delete_all
+```
+
+4) Delete checksums
+
+```ruby
+rails> Stats::StatsReport.update_all(as_document_checksum: nil)
+```
```

Comment:
is there a way to have a `rails 'storage:rollback[stats_reports]'` task that encapsulates this logic in the event we need to do it.